plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1397,6 +1404,47 @@ public void setColumnComment(ConnectorSession session, ConnectorTableHandle tabl
         return viewToken;
     }
 
+    @Override
+    public Optional<CatalogSchemaTableName> redirectTable(ConnectorSession session, SchemaTableName tableName)
+    {
+        return tableRedirectionProvider.redirectTable(session, tableName);
+    }
+
+    @Override
+    public Stream<TableColumnsMetadata> streamTableColumns(ConnectorSession session, SchemaTablePrefix prefix)
+    {
+        requireNonNull(prefix, "prefix is null");
+        List<SchemaTableName> schemaTableNames;
+        if (prefix.getTable().isEmpty()) {
+            schemaTableNames = catalog.listTables(session, prefix.getSchema());
+        }
+        else {
+            SchemaTableName schemaTableName = prefix.toSchemaTableName();
+            try {
+                catalog.loadTable(session, schemaTableName);
```

Comment:
looks unused, right?

Reply:
Yes, the result of the statement is indeed unused. I want to verify whether I'm dealing with a valid iceberg table or not and this is why I am making the method call.
`TrinoCatalog` does not expose at the moment a method for handling this need.

Reply:
I moved the handling for `UnknownTableTypeException` below due to https://github.com/trinodb/trino/pull/11356/files/af21763756e96b7981d18d33f4570b472e7c6fa1..909f1cc3caf3c4a289dfd7927eee66508a718093#r831213637 so this cumbersome handling is not needed anymore.